ui/trace_window.cpp

```diff
@@ -228,26 +245,44 @@ void TraceDialog::UpdateDeviceList()
     m_devices = cur_list;
     m_dev_model->clear();
 
-    for (size_t i = 0; i < m_devices.size(); i++)
+    if (m_devices.size() == 0)
```

Comment:
same as above

Reply:
actually this code block looks like a copy of the above, can we put it in a function instead of duplicating the code. (a bit error prone this way, people need to modify code in both places if any modification is required. )